systemd/systemd.go

```diff
@@ -213,17 +225,28 @@ func NewCollector(logger log.Logger) (*Collector, error) {
 		ipEgressBytes:                 ipEgressBytes,
 		ipIngressPackets:              ipIngressPackets,
 		ipEgressPackets:               ipEgressPackets,
+		scrapeSuccessDesc:             scrapeSuccessDesc,
+		scrapeDurationDesc:            scrapeDurationDesc,
 		unitIncludePattern:            unitIncludePattern,
 		unitExcludePattern:            unitExcludePattern,
 	}, nil
 }
 
 // Collect gathers metrics from systemd.
 func (c *Collector) Collect(ch chan<- prometheus.Metric) {
+	begin := time.Now()
+
 	err := c.collect(ch)
 	if err != nil {
 		level.Error(c.logger).Log("msg", "error collecting metrics", "err", err)
+		ch <- prometheus.MustNewConstMetric(
+			c.scrapeSuccessDesc, prometheus.GaugeValue, 0, namespace)
+	} else {
+		ch <- prometheus.MustNewConstMetric(
+			c.scrapeSuccessDesc, prometheus.GaugeValue, 1, namespace)
 	}
+
+	ch <- prometheus.MustNewConstMetric(c.scrapeDurationDesc, prometheus.GaugeValue, time.Since(begin).Seconds(), namespace)
```

Comment:
This doesn't make sense to me, as it's not actually exposing metrics per collector that I can tell. `namespace` is always `systemd`, so the label would always be `collector="systmed"`.

This means it's no different than `up` and `scrape_duration_seconds` recorded by Prometheus.